kore/src/Kore/Step/Axiom/Identifier.hs

```diff
@@ -88,18 +98,19 @@ Returns 'Nothing' if the 'TermLike' does not conform to one of the structures we
 recognize.
 -}
 matchAxiomIdentifier ::
-    InternalVariable variable =>
     TermLike variable ->
     Maybe AxiomIdentifier
-matchAxiomIdentifier = Recursive.fold matchWorker . Builtin.externalize
+matchAxiomIdentifier = Recursive.fold matchWorker
   where
-    matchWorker (_ :< patternF) =
-        case patternF of
-            ApplicationF application ->
-                pure (Application symbolId)
-              where
-                symbol = Syntax.applicationSymbolOrAlias application
-                symbolId = Syntax.symbolOrAliasConstructor symbol
+    matchWorker (_ :< termLikeF) =
+        case termLikeF of
+            ApplyAliasF application ->
+                pure $
+                    Application $
+                        Syntax.symbolOrAliasConstructor $
+                            Alias.toSymbolOrAlias $
+                                Syntax.applicationSymbolOrAlias application
+            ApplySymbolF application -> mkAppId application
```

Comment:
Why does the case for `ApplyAliasF` get inlined, but not `ApplySymbolF`?

Reply:
I was originally gonna leave them both inlined but then realised that I can use the `mkAppId` that I defined for other stuff to simplify the case for `ApplySymbolF`. But I'll add another function to encapsulate the Alias case as well if you feel like that would be better.